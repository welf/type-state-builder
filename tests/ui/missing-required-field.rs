use type_state_builder::TypeStateBuilder;

#[derive(TypeStateBuilder)]
struct User {
    #[builder(required)]
    name: String,

    #[builder(required)]
    email: String,

    age: Option<u32>,
}

fn main() {
    // This should fail: trying to build without setting the required 'email' field
    let user = User::builder()
        .name("Alice".to_string())
        .build(); // Error: missing required field 'email'
}
