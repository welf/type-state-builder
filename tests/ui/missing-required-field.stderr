error[E0599]: no method named `build` found for struct `UserBuilder_HasName_MissingEmail` in the current scope
  --> tests/ui/missing-required-field.rs:18:10
   |
3  |   #[derive(TypeStateBuilder)]
   |            ---------------- method `build` not found for this struct
...
16 |       let user = User::builder()
   |  ________________-
17 | |         .name("Alice".to_string())
18 | |         .build(); // Error: missing required field 'email'
   | |         -^^^^^ method not found in `UserBuilder_HasName_MissingEmail`
   | |_________|
   |
